pkg/controller/directvolumemigration/task.go

```diff
@@ -222,7 +222,7 @@ func (t *Task) Run(ctx context.Context) error {
 		// destination
 		err := t.ensureDestinationNamespaces()
 		if err != nil {
-			liberr.Wrap(err)
+			return liberr.Wrap(err)
```

Comment:
will this not fail when the namespace already exists?

Reply:
@alaypatel07 right, I missed that. adding an error check in the function

Reply:
@alaypatel07 added the check

Reply:
@pranavgaikwad +1, I think there is another check to be made here, i.e. if namespace.finalizer is set. If that is set the namespace is in delete mode i.e. it will be deleted soon when the itinerary is in the next steps